### PR TITLE
Robustify teardown in storage tests + fix reboot issue in MooseFS tests

### PR DESCRIPTION
**Pull request by stormi**
Several commits, possibly better reviewed separately.

### CODE REVIEW

File: lib/pool.py [hunk 1]
```diff
@@ -19,6 +20,58 @@ def __init__(self, master_hostname_or_ip):
         self.uuid = self.master.xe('pool-list', minimal=True)
         self.saved_uefi_certs = None
 
+    def exec_on_hosts_on_error_rollback(self, func, rollback_func, host_list=[]):
+        """
+        Execute a function on all hosts of the pool.
+
+        If something fails, and execute rollback_func on each host on which the command was already executed.
```
Comment (benjamreis) [inline lib/pool.py:27]:
```suggestion
        If something fails: execute rollback_func on each host on which the command was already executed.
```

File: lib/pool.py [hunk 2]
```diff
@@ -19,6 +20,58 @@ def __init__(self, master_hostname_or_ip):
...
+                        self.exec_on_hosts_on_error_continue(rollback_func, rollback_hosts)
+                    except Exception:
+                        pass
+                raise e
+
+    def exec_on_hosts_on_error_continue(self, func, host_list=[]):
+        """
+        Execute a function on all hosts of the pool.
+
+        If something fails, store the exception but still attempt the function on the next hosts.
```
Comment (benjamreis) [inline lib/pool.py:57]:
```suggestion
        If something fails: store the exception but still attempt the function on the next hosts.
```

File: lib/sr.py [hunk 3]
```diff
@@ -14,17 +14,27 @@ def __init__(self, uuid, pool):
...
+        ))[0]
+
+    def unplug_pbd(self, pbd_uuid, force=False):
+        try:
+            self.pool.master.xe('pbd-unplug', {'uuid': pbd_uuid})
+        except commands.SSHCommandFailed as e:
+            # We must be sure to execute correctly "unplug" on unplugged VDIs without error
+            # if force is set.
+            if not force:
+                raise
```
Comment (benjamreis) [inline lib/sr.py:31]: The comment and code are contradictary

File: lib/sr.py [hunk 4]
```diff
@@ -33,10 +43,13 @@ def all_pbds_attached(self):
...
         return all_attached
 
+    def plug_pbd(self, pbd_uuid):
+        self.pool.master.xe('pbd-plug', {'uuid': pbd_uuid})
+
     def plug_pbds(self, verify=True):
         logging.info("Attach PBDs")
         for pbd_uuid in self.pbd_uuids():
-            self.pool.master.xe('pbd-plug', {'uuid': pbd_uuid})
+            self.plug_pbd(pbd_uuid)
```
Comment (benjamreis) [inline lib/sr.py:52]: Would it make sense to make a `PBD` API?

**Review by benjamreis — CHANGES_REQUESTED** [review]

**Review by gduperrey — APPROVED** [review]

**Comment by stormi** [comment]
I added another commit that depends on the previous ones.

File: lib/sr.py (hunk 3 above)
Comment (benjamreis) [reply lib/sr.py:31]: My bad, you're right

**Review by benjamreis — APPROVED** [review]

**Review by Wescoeur — APPROVED** [review]

**Pull request merged**